src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -41,12 +42,27 @@ internal class UnixPwmChannel : PwmChannel
             Validate();
             Open();
 
-            // avoid opening the file for operations changing relatively frequently
-            _dutyCycleWriter = new StreamWriter(new FileStream($"{_channelPath}/duty_cycle", FileMode.Open, FileAccess.ReadWrite));
-            _frequencyWriter = new StreamWriter(new FileStream($"{_channelPath}/period", FileMode.Open, FileAccess.ReadWrite));
+            var timeout = OpenTimeout;
+            // If we're running as non-root (and the udev rule needs to kick in here) we 
+            // might need a few retries until we can open the channel files. 
+            while (timeout > TimeSpan.Zero)
```

Comment:
Not really sure why you used a TimeSpan here, seems like what you are trying to do would be more appropiate to use with a `numberOfRetries` or something like that, where we would loop over a specified number of retries with a sleep in between each retry.

Is this only needed for non-root access? If so, I wouldn't add this code to optimize for that as I have already pointed out since we rather require root access for running, so I would just remove this whole logic.

Reply:
Unfortunately, non-root access doesn't work without this (because it fails all the time). 
I think waiting for a certain time is better than a number of retries, since it takes a certain time for the rights to be set up. I don't have a big preference there, though.

Reply:
I think I didn't fully explained my point, what I meant is that you don't want to use TimeSpan as the condition in your while, instead you would probably want to do a retry number as there is no reason why to use TimeSpan. Like:

```c#
int numRetries = 5;
while (numRetries--)
{
  try
  {
    // do stuff.
  }
  catch (Exception e)
  {
    if (numRetries == 0)
    {
      throw new AggregateException("Exceeded the number of retries", e)
    }
    Thread.Sleep(someReasonableTimeLikeAMillisecondOrSo);
  }
}
```

Again, I don't think we want to optimize for non-root access, so if all of this code is just for enabling that, then I would just remove it.

Reply:
Both loops do about the same on the bottom line, but if you think your's is easier to read, I can change that. 
I do think that supporting non-root access is an important feature that will be requested by many users (in general of course, not only for this part).

Reply:
I totally agree that non-root access would be awesome to have, but the current way we have for doing it is not good. Specially since we already said that we won't even copy the file over so that means that users would really still have to do some manual work in order to enable this. If that's the case, I much rather have a readme somewhere in our repo indicating which file you have to copy, and how to add the retries on your code in order to enable non-root access. If we were to have a great solution for this, then I would most definitely not be pushing back here.